CRM/Core/BAO/Setting.php

```diff
@@ -178,6 +178,9 @@ public static function setItem(
     /** @var \Civi\Core\SettingsManager $manager */
     $manager = \Civi::service('settings_manager');
     $settings = ($contactID === NULL) ? $manager->getBagByDomain($domainID) : $manager->getBagByContact($domainID, $contactID);
+    if (self::isEnvironmentSet($name)) {
+      throw new api_Exception('CiviCRM Environment already set in civicrm.settings.php!');
```

Comment:
REST API returns error=1.
There would be no need for a UI check since we are disabling the widget if already set in civicrm.settings.php